File: Algorithm/Sort/BubbleSort/BubbleSort.py
#!/usr/local/bin/python3
def bubble_sort(items):
    if len(items) <= 1:
        return items
    for index in range(len(items)-1,0,-1):
        flag = False
        for sub_index in range(index):
            if items[sub_index] > items[sub_index + 1]:
                items[sub_index],items[sub_index + 1] = items[sub_index + 1], items[sub_index]
                flag = True
        if flag == False:
            break
    return items


testlist = [17, 23, 20, 14, 12, 25, 1, 20, 81, 14, 11, 12]
print('Before selection sort: {}'.format(testlist))
print('After selection sort:  {}'.format(bubble_sort(testlist)))

File: Algorithm/Sort/BubbleSort/BubbleSort.cpp
#include <iostream>
#include <algorithm>
using namespace std;

void outputArray(int arr[],int len) {
    while (len-- >0) {
        printf("%d ",*arr++);
    }
    printf("\n");
}

void bubble_sort(int arr[], int len) {
	for (int i = 0; i < len - 1; i++)
		for (int j = 0; j < len - 1 - i; j++)
			if (arr[j] > arr[j + 1])
				swap(arr[j], arr[j + 1]);
}

int main() {
  int a[] = {5, 9, 20, 31, 8, 89, 1, 15, 100, 2};
  cout<<"before sorting"<<endl;
  outputArray(a, 10);
  cout<<"after sorting"<<endl;
  bubble_sort(a,10);
  outputArray(a, 10);

  return 0;
}
